py/objtype.c

```diff
@@ -888,6 +890,7 @@ mp_obj_t mp_obj_new_type(qstr name, mp_obj_t bases_tuple, mp_obj_t locals_dict)
     mp_obj_tuple_get(bases_tuple, &len, &items);
     for (uint i = 0; i < len; i++) {
         assert(MP_OBJ_IS_TYPE(items[i], &mp_type_type));
+#if 0
```

Comment:
Whether it's needed or not depends on https://github.com/micropython/micropython/issues/2092 (if needed, some other code is needed)